### PR TITLE
Fix functionality with confirmation user email 

### PR DESCRIPTION
**Pull request by mvmoshchovskyi**
fixed bug durinr email confirm
595-email
595-jwt token
595-user

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Review by grebiniuk — CHANGES_REQUESTED** [review]
Please fix the code smell

**Review by AndriyLymych — APPROVED** [review]

File: src/redux/cart/cart.operations.js [hunk 1]
```diff
@@ -107,6 +107,13 @@ const getCartByUserId = async (userId) => {
   `;
   const result = await getItems(getCartByUserIdQuery, { id: userId });
 
+  if (
+    result?.data?.getCartByUserId?.message &&
+    result?.data?.getCartByUserId?.message !== 'CART_IS_NOT_FOUND'
```
Comment (AndriyLymych) [inline src/redux/cart/cart.operations.js:112]: remove `result?.data?.getCartByUserId?.message !== 'CART_IS_NOT_FOUND'` this useless check

**Review by AndriyLymych — DISMISSED** [review]

File: src/redux/cart/cart.operations.js [hunk 2]
```diff
@@ -107,6 +107,10 @@ const getCartByUserId = async (userId) => {
   `;
   const result = await getItems(getCartByUserIdQuery, { id: userId });
 
+  if (result?.data?.getCartByUserId?.message) {
```
Comment (grebiniuk) [inline src/redux/cart/cart.operations.js:110]: You don't need this.
getItes already throw the error with the message.
It should be handled in sagas error handler.

File: src/redux/user/user.operations.js [hunk 3]
```diff
@@ -106,7 +106,9 @@ const loginUser = async (data) => {
 }
   `;
   const result = await setItems(loginUserMutation, data);
-
+  if (result?.data?.loginUser?.message) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:109]: the same, you don't need this check

File: src/redux/user/user.operations.js [hunk 4]
```diff
@@ -131,14 +133,21 @@ const getGoogleUser = async (payload) => {
...
+        token
+        refreshToken
+        confirmed
+    }
   }
   `;
-  const result = await setItems(confirmUserEmailMutation, { data });
+  const result = await setItems(confirmUserEmailMutation, { token });
 
+  if (result?.data?.confirmUserEmail?.message) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:148]: the same

File: src/redux/user/user.operations.js [hunk 5]
```diff
@@ -176,7 +185,9 @@ const registerUser = async ({ user, language }) => {
         }
       `;
   const result = await setItems(registerUserMutation, { user, language });
-
+  if (result?.data?.registerUser?.message) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:188]: same

File: src/redux/user/user.operations.js [hunk 6]
```diff
@@ -222,18 +233,22 @@ const updateUserById = async ({ user, id, upload }) => {
     }
   `;
   const result = await setItems(updateUserByIdMutation, { user, id, upload });
-
+  if (result?.data?.updateUserById?.message) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:236]: same

File: src/redux/user/user.operations.js [hunk 7]
```diff
@@ -222,18 +233,22 @@ const updateUserById = async ({ user, id, upload }) => {
...
+const sendEmailConfirmation = async ({ email, language }) => {
   const sendEmailConfirmationMutation = `
      mutation sendConfirmation($email: String!, $language: Int!){
       sendEmailConfirmation(email: $email, language: $language)
     }
   `;
-  const result = await setItems(sendEmailConfirmationMutation, { data });
-
+  const result = await setItems(sendEmailConfirmationMutation, { email, language });
+  if (result?.data?.sendEmailConfirmation?.message) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:249]: same

File: src/redux/user/user.operations.js [hunk 8]
```diff
@@ -304,6 +319,10 @@ const getUserByToken = async () => {
 
   const result = await getItems(getUserByTokenQuery);
 
+  if (result?.data?.getUserByToken?.statusCode) {
```
Comment (grebiniuk) [inline src/redux/user/user.operations.js:322]: same

File: src/redux/user/user.reducer.js [hunk 9]
```diff
@@ -26,7 +27,8 @@ export const initialState = {
   passwordReset: false,
   confirmationEmailSent: false,
   recoveryLoading: false,
-  confirmationLoading: false
+  confirmationLoading: false,
+  emailError: null
```
Comment (grebiniuk) [inline src/redux/user/user.reducer.js:31]: that is not a good idea to add an additional error field

File: src/redux/user/user.sagas.js [hunk 10]
```diff
@@ -273,6 +277,15 @@ export function* handleUserError(e) {
   }
 }
 
+export function* handleEmailError(e) {
```
Comment (grebiniuk) [inline src/redux/user/user.sagas.js:280]: Looks like an additional state reducer and handler for email error is unnesesery

**Review by grebiniuk — CHANGES_REQUESTED** [review]

File: src/redux/cart/cart.operations.js (hunk 2 above)
Comment (mvmoshchovskyi) [reply src/redux/cart/cart.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 3 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 4 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 5 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 6 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 7 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.operations.js (hunk 8 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.operations.js]: fixed

File: src/redux/user/user.reducer.js (hunk 9 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.reducer.js]: fixed

File: src/redux/user/user.sagas.js (hunk 10 above)
Comment (mvmoshchovskyi) [reply src/redux/user/user.sagas.js]: fixed

**Review by grebiniuk — APPROVED** [review]

**Pull request merged**